enterprise/cloud.amazon/src/org/netbeans/modules/cloud/amazon/AmazonInstanceManager.java

```diff
@@ -113,7 +116,7 @@ private void store(AmazonInstance ai) {
                 continue;
             }
             String key = new String(ch);
-            result.add(new AmazonInstance(name, keyId, key, region));
+            result.add(new AmazonInstance(name, keyId, key, region, code));
```

Comment:
Just to make sure - does this work if code is null? That situation would be reached when an old userdir is read with a new version of netbeans.

Reply:
Great catch.  I didn't even think of this :(

We need to use the region code as setting a region endpoint by URL is deprecated in newer AWS SDK, and won't work for newer regions (e.g. London).

So I've broken the details of the Amzon regions out from the Wizard component into a seperate AmazonRegion enum.  Then I'm checking in AmazonInstanceManager if code is null, and if it is, I can retrieve it from the enum based on the URL.  I beleive that will mean old userdirs containing just the URL won't break with the new versions of the modules...